src/Avalonia.Controls/ApplicationLifetimes/ClassicDesktopStyleApplicationLifetime.cs

```diff
@@ -123,6 +129,14 @@ public void Dispose()
             if (_activeLifetime == this)
                 _activeLifetime = null;
         }
+        
+        private void ShutdownRequested(object sender, CancelEventArgs e)
+        {
+            foreach (var w in Windows.ToArray())
```

Comment:
Just curious. What is the purpose of calling ToArray() here?

Reply:
Inside the event, the list of windows could be modified. Or even a new window opened, so you need to create a snap shot for it to work correctly.

Reply:
Look at the Windows property. You already receive a snapshot from this property when you start iterating.

Reply:
Y looks like the ToArray isn't needed

Reply:
Good catch, I'd not noticed that - fixed.